src/main/scala/io/iohk/ethereum/rlp/package.scala

```diff
@@ -20,10 +20,14 @@ package object rlp {
 
   def encode[T](input: T)(implicit enc: RLPEncoder[T]): Array[Byte] = RLP.encode(enc.encode(input))
 
+  def encodeToRlp[T](input: T)(implicit enc: RLPEncoder[T]): RLPEncodeable = enc.encode(input)
+
   def encode(input: RLPEncodeable): Array[Byte] = RLP.encode(input)
 
   def decode[T](data: Array[Byte])(implicit dec: RLPDecoder[T]): T = dec.decode(RLP.rawDecode(data))
 
+  def decode[T](data: RLPEncodeable)(implicit dec: RLPDecoder[T]): T = dec.decode(data)
```

Comment:
was able to remove `encodeToRlp`, only thing left is 
`def decode[T](data: RLPEncodeable)(implicit dec: RLPDecoder[T]): T` which has different argument type than other `decode` do you thing it is necessary to indicate that it decodes from RLP if it is type of input argument? 
Besides that it does the same thing as 
`def decode[T](data: Array[Byte])(implicit dec: RLPDecoder[T]): T`